src/main/java/com/bortbort/arduino/FiloFirmata/Parser/CommandParser.java

```diff
@@ -18,16 +16,13 @@
 public class CommandParser {
     private static final Logger log = LoggerFactory.getLogger(CommandParser.class);
 
-    /**
-     * Map of messageBuilders used to parse an array/inputstream of bytes into a Firmata Message object.
-     */
-    private static final HashMap<Byte, MessageBuilder> messageBuilderMap = new HashMap<>();
+    public static final CommandParserInstance DEFAULT_INSTANCE = new CommandParserInstance(log);
 
     /**
      * Register all pre-built Message builder objects available in the Firmata library.
      */
     static {
-        CommandParser.addParser(
+        DEFAULT_INSTANCE.addParser(
```

Comment:
Shouldn't the commands in this class be referencing some sort of CommandParserInstance currentInstance? Perhaps passed 
 as a reference to the configuration instance object that is returned when calling configuration.getCommandParserInstance() ? 
You may also need to split out the static{} call and turn it into an array of parsers that can be added whenever the currentInstance is set or changed. 

As it is written right now there is no way to actually configure a separate instance for use until some sort of tracking of the 'intended' instance is added into CommandParser.java